torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -0,0 +1,357 @@
+
+
+def should_pad_bmm(match):
+    mat1, mat2 = fetch_fake_tensors(match, ("mat1", "mat2"))
+    return (
+        torch._inductor.config.shape_padding
+        and check_device(mat1, mat2)
+        and not is_symbolic(mat1, mat2)
+        and should_pad_bench(mat1, mat2, torch.ops.aten.bmm)
+    )
+
+
+def bmm_replace(mat1, mat2):
+    m_padded_length = get_padded_length(mat1.shape[1], get_alignment_size(mat1))
+    k_padded_length = get_padded_length(mat1.shape[2], get_alignment_size(mat1))
+    n_padded_length = get_padded_length(mat2.shape[2], get_alignment_size(mat2))
+
+    if m_padded_length != 0 or k_padded_length != 0 or n_padded_length != 0:
+        return pad_bmm(mat1, mat2, m_padded_length, k_padded_length, n_padded_length)
+
+    return aten.bmm(mat1, mat2)
+
+
+def pad_bmm(mat1, mat2, m_padded_length, k_padded_length, n_padded_length):
+    # bmm_replace will go through pad_bmm multiple times if multiple dimensions are needed to be padded
+    if k_padded_length != 0:
+        mat1 = pad_dim(mat1, k_padded_length, 2)
+        mat2 = pad_dim(mat2, k_padded_length, 1)
+
+        return aten.ops.bmm(mat1, mat2)
+    elif n_padded_length != 0:
+        mat2 = pad_dim(mat2, n_padded_length, 2)
+        return aten.ops.bmm(mat1, mat2)[:, :, :-n_padded_length].contiguous()
+    else:
+        mat1 = pad_dim(mat1, m_padded_length, 1)
+        return aten.ops.bmm(mat1, mat2)[:, :-m_padded_length, :].contiguous()
+
+
+@functools.lru_cache(None)
+def _pad_mm_init():
+    from .joint_graph import patterns
+
+    if torch.cuda.is_available():
+        # workaround https://github.com/pytorch/pytorch/issues/97894
+        device = "cuda"
+    else:
+        device = "cpu"
+
+    # sizes/values dont actually matter for initial trace
+    # once we get a possible match we re-trace with the actual values and verify the match still holds
+
+    dim2a = functools.partial(torch.empty, (4, 4), device=device, requires_grad=True)
+    dim2b = functools.partial(torch.empty, (4, 4), device=device, requires_grad=True)
+
+    dim3a = functools.partial(torch.empty, (4, 4, 4), device=device, requires_grad=True)
+    dim3b = functools.partial(torch.empty, (4, 4, 4), device=device, requires_grad=True)
+
+    dim1a = functools.partial(torch.empty, (4), device=device, requires_grad=True)
+
+    # workaround https://github.com/pytorch/pytorch/issues/97894
+    # 0.113377 is a "magic" value that lets us recover the lost input arg relationship
+    rep = {"beta": 0.213377, "alpha": 0.113377}
+
+    for pattern, replacement, args, workaround, extra_check in [
+        (
+            mm_pattern,
+            mm_replace,
+            [dim2a(), dim2b()],
+            {},
+            should_pad_mm,
+        ),
+        (
+            bmm_pattern,
+            bmm_replace,
+            [dim3a(), dim3b()],
+            {},
+            should_pad_bmm,
+        ),
+        (
+            addmm_pattern,
+            addmm_replace,
+            [dim1a(), dim2a(), dim2b()],
```

Comment:
cc @jansel anything obviously wrong here or pointers on what I should change ? 

Getting 
```
  File "/scratch/eellison/work/pytorch/torch/_inductor/compile_fx.py", line 690, in partition_fn
    joint_graph_passes(graph)
  File "/scratch/eellison/work/pytorch/torch/_inductor/fx_passes/joint_graph.py", line 38, in joint_graph_passes
    count = patterns.apply(graph.graph)
  File "/scratch/eellison/work/pytorch/torch/_inductor/pattern_matcher.py", line 764, in apply
    if m and entry.extra_check(m):
  File "/scratch/eellison/work/pytorch/torch/_inductor/pattern_matcher.py", line 634, in check_fn
    [match.kwargs[name] for name in argnames], lambda n: n.meta["val"]
  File "/scratch/eellison/work/pytorch/torch/_inductor/pattern_matcher.py", line 634, in <listcomp>
    [match.kwargs[name] for name in argnames], lambda n: n.meta["val"]
torch._dynamo.exc.BackendCompilerFailed: backend='compile_fx_wrapper' raised:
KeyError: 'beta'
```  

Here for addmm

Reply:
Nvm i forgot to pass in scalar_workaround